dataset/src/test/scala/frameless/ops/PivotTest.scala

```diff
@@ -0,0 +1,62 @@
+package frameless
+package ops
+
+import org.scalacheck.{Gen, Prop}
+import org.scalacheck.Prop._
+import shapeless._
+import frameless.functions.aggregate._
+import org.apache.spark.sql.{functions => sparkFunctions}
+
+class PivotTest extends TypedDatasetSuite {
+  def withCustomGenX4: Gen[Vector[X4[String, String, Int, Boolean]]] = {
+    val kvPairGen: Gen[X4[String, String, Int, Boolean]] = for {
+      a <- Gen.oneOf(Seq("1", "2", "3", "4"))
+      b <- Gen.oneOf(Seq("a", "b", "c"))
```

Comment:
I would expect the custom gen to be only constrained on the pivot values, why do you need the other ones?